Provide documentation for new custom avatars feature (#127)
gitextensions/GitExtensionsDoc: source/settings.rst

    .. settingsgroup:: Author images
      :id: author-images
      
      .. setting:: Show author's avatar column in the commit graph
        :id: show-avatar-commit-graph

        If checked, avatar images are downloaded for commit authors and shown on the right side in the commit graph.

      .. setting:: Show author's avatar in the commit info view
        :id: show-avatar-commit-info

        If checked, avatar images are downloaded for commit authors and shown in the commit info view.

      .. setting:: Avatar provider
        :id: avatar-provider

        The avatar provider setting determines the source from which avatar images are requested.

>         - ``Default`` - The default avatar provider loads a user defined avatar images, depending on the email address, from GitHub or Gravatar.
>           If no user defined image could be found, a fallback images is used.

Reviewer: Describe the algorithm more in detail?

```suggestion
        - ``Default`` - The default avatar provider loads a user defined avatar images, depending on the email address, from GitHub 
          (if this is a GitHub repo and the email is ending with @users.noreply.github.com), 
          otherwise query `gravatar <http://gravatar.com/>`.
          Note that GitHub users may configure the account to use Gravatar too.
          If no user defined image could be found, a fallback images is used.
```

Author: I'm not sure if the exact details should be in the user manual, because it would be outdated as soon as someone changes a few things.

There are also quite a few edge cases that might be pretty hard (or unnecessary) do describe. For example the bot account detection for github email addresses and the fact that if a github user decides to use gravatar (in the github settings) we still get the avatar from github because github loads it from gravatar and caches it (and it is also a request less for us if the first one already succeeds)

        - ``None`` - If selected, no user-defined images are loaded and the fallback is evaluated immediately.
        - ``Custom`` - An advanced mode that allows you to set one or more custom avatar provider services (e.g. Libravatar) by providing URL
          templates.

          URL Template Syntax
            The URL template syntax consists of regular URLs to avatar images, that can be enriched with variables, which are substituted before
            evaluation. Those variables are encoded using curly brackets ``{}`` and can be used like this: ``https://example.avatar.service/u/{email}/avatar.png``.
            If a request fails (http 400 and 500 errors) or does not provide a valid image, the next URL is used. More URLs can be specified by chaining them
            together with semicolons (";") like so: ``http://provider1.com/{sha1}.png;http://provider2.com/{sha1}.png``. If all custom URLs fail to provide

>             an avatar image, the gitextensions internal fallback mechanism will provide one for that user.

Reviewer 2: ```suggestion
            an avatar image, the application internal fallback mechanism will provide one for that user.
```
or
```suggestion
            an avatar image, the Git Extensions internal fallback mechanism will provide one for that user.
```

            with an empty string, so the resulting URL never contains the curly brackets.

            The following variables are currently supported:

            - ``name`` - The name of the commit author (git config ``user.name``). Special characters are URL encoded.
            - ``email`` - The email address of the commit author (git config ``user.email``). Special characters are URL encoded.
            - ``md5`` - A lowercase hex representation of the MD5 hash of the normalized (all characters lowercase)
              email address (without URL encoding). This hash is compatible with Gravatar and thus compatible with a lot of similar services.
            - ``sh1`` - Like the ``md5`` variable but with SHA1 as hash algorithm.
            - ``sha256`` - Like the ``md5`` variable but with SHA256 as hash algorithm.
            - ``imagesize`` - Represents the requested avatar size in pixels.

            A complete working configuration might look something like this:
            ``https://www.libravatar.org/avatar/{md5}?s={imageSize}&default=404;https://avatar.tobi.sh/{md5}?size={imageSize}``

      .. setting:: Fallback generated avatar style
        :id: avatar-fallback

>         The configured fallback determines how authors without a user-defined avatar are rendered. Besides ``Author Initials`` all other options are

Reviewer 2: ```suggestion
        The configured fallback determines how authors without a user-defined avatar are presented. Besides ``Author Initials`` all other options are
```

>         provided by Gravatar. Details about their fallback modes can be found here https://en.gravatar.com/site/implement/images/ in the section "Default Image".

Reviewer: Is only Gravatar using other than Author Initials? How to rewrite this?

Author: Sorry I don't quite get what you mean. Of the currently available / implemented fallback options
![image](https://user-images.githubusercontent.com/2495113/106068213-825d9f00-6100-11eb-958b-29fa053f302e.png)
only `Author Initials` is done locally (by gitextensions)

For the other options we query gravatar. Or strictly speaking we make sure that gravatar is queried with a set fallback (so we don't have to send a query to gravatar twice)

>         ``Author Initials`` are generated by gitextensions internally and require no network connection to be displayed.

Reviewer 2: ditto
```suggestion
        ``Author Initials`` are generated by application internally and require no network connection to be displayed.
```

Reviewer: the application
